lib/_http_client.js

```diff
@@ -801,13 +801,15 @@ function onSocketNT(req, socket, err) {
       req.emit('close');
     }
 
-    if (!err && req.agent) {
-      socket?.emit('free');
-    } else if (socket) {
-      finished(socket.destroy(err || req[kError]), (er) => {
-        _destroy(req, er || err);
-      });
-      return;
+    if (socket) {
+      if (!err && req.agent && !socket.destroyed) {
```

Comment:
@lpinca `!socket.destroyed` is the primary fix which also ensure the error is forwarded.